mdata/cache/ccache_metric.go

```diff
@@ -401,15 +415,22 @@ func (mc *CCacheMetric) Search(ctx context.Context, metric schema.AMKey, res *CC
 	}
 
 	if !res.Complete && res.From > res.Until {
-		log.Debug("CCacheMetric Search: Found from > until (%d/%d), printing chunks\n", res.From, res.Until)
-		mc.debugMetric()
+		log.Warn("CCacheMetric Search: Found from > until (%d/%d), key = %s, printing chunks\n", res.From, res.Until, mc.MKey.String())
+		mc.debugMetric(from-7200, until+7200)
+		res.Complete = false
+		res.Start = res.Start[:0]
+		res.End = res.End[:0]
+		res.From = from
+		res.Until = until
 	}
 }
 
-func (mc *CCacheMetric) debugMetric() {
-	log.Debug("CCacheMetric debugMetric: --- debugging metric ---\n")
+func (mc *CCacheMetric) debugMetric(from, until uint32) {
+	log.Warn("CCacheMetric debugMetric: --- debugging metric between %d and %d ---\n", from, until)
 	for _, key := range mc.keys {
-		log.Debug("CCacheMetric debugMetric: ts %d; prev %d; next %d\n", key, mc.chunks[key].Prev, mc.chunks[key].Next)
+		if key >= from && key <= until {
+			log.Warn("CCacheMetric debugMetric: ts %d; prev %d; next %d\n", key, mc.chunks[key].Prev, mc.chunks[key].Next)
+		}
 	}
-	log.Debug("CCacheMetric debugMetric: ------------------------\n")
+	log.Warn("CCacheMetric debugMetric: ------------------------\n")
```

Comment:
a concern here could be that log lines from different requests may interleave, or that the logs may be interleaved with unrelated logs. but if it's ok with you it's ok with me

Reply:
Yes, that definitely happened. I used grep since all the lines starting with "CCacheMetric" makes it easy